modules/github-oidc-role/main.tf

```
@@ -1,29 +1,24 @@
 locals {
-  enabled             = module.this.enabled
-  aws_canned_policies = [for arn in var.aws_iam_policies : arn if can(regex("^arn:aws:iam::aws:policy/", arn))]
-  aws_policies        = length(local.aws_canned_policies) > 0 ? local.aws_canned_policies : null
+  enabled          = module.this.enabled
+  managed_policies = [for arn in var.iam_policies : arn if can(regex("^arn:aws[^:]*:iam::aws:policy/", arn))]
+  policies         = length(local.managed_policies) > 0 ? local.managed_policies : null
   policy_document_map = {
     "gitops"        = local.gitops_policy
     "lambda_cicd"   = local.lambda_cicd_policy
     "inline_policy" = one(module.iam_policy.*.json)
   }
-  active_policy_map = { for k, v in local.policy_document_map : k => v if v != null }
+  custom_policy_map = merge(local.policy_document_map, local.overridable_additional_custom_policy_map)
+
+  active_policy_map = { for k, v in local.custom_policy_map : k => v if v != null }
```

We have seen "gotchas" where the policy ends up being empty instead of null, the empty policy being `"{}"`, so maybe
```suggestion
  # Ignore empty policies of the form `"{}"` as well as null policies
  active_policy_map = { for k, v in local.custom_policy_map : k => v if try(length(v), 0) > 3 }
```